tests/test_awkward.py

```diff
@@ -2,15 +2,50 @@
 
 import pylhe
 
-TEST_FILE = skhep_testdata.data_path("pylhe-testfile-pr29.lhe")
+TEST_FILE_WITHOUT_WEIGHTS = skhep_testdata.data_path("pylhe-testfile-pr29.lhe")
+TEST_FILE_WITH_WEIGHTS = skhep_testdata.data_path("pylhe-testlhef3.lhe")
 
 
 def test_to_awkward():
-    arr = pylhe.to_awkward(pylhe.read_lhe_with_attributes(TEST_FILE))
+    arr = pylhe.to_awkward(pylhe.read_lhe_with_attributes(TEST_FILE_WITHOUT_WEIGHTS))
     assert len(arr) == 791
-    assert len(arr.eventinfo.nparticles) == len(arr.particles)
+    assert len(arr) == len(arr.particles)
+    assert len(arr) == len(arr.eventinfo)
+    for field in arr.particles.fields:
+        assert len(arr) == len(arr.particles[field])
+    for field in arr.eventinfo.fields:
+        assert len(arr) == len(arr.eventinfo[field])
+    assert "weights" not in arr.fields
+
+    arr = pylhe.to_awkward(pylhe.read_lhe_with_attributes(TEST_FILE_WITH_WEIGHTS))
+    assert len(arr) == 59
+    assert len(arr) == len(arr.particles)
+    assert len(arr) == len(arr.eventinfo)
+    for field in arr.particles.fields:
+        assert len(arr) == len(arr.particles[field])
+    for field in arr.eventinfo.fields:
+        assert len(arr) == len(arr.eventinfo[field])
+    assert "weights" in arr.fields
+    assert len(arr) == len(arr.weights)
+    assert arr.weights.fields == [
+        "1001",
+        "1002",
+        "1003",
+        "1004",
+        "1005",
+        "1006",
+        "1007",
+        "1008",
+        "1009",
+    ]
```

Comment:
```suggestion
    ]
    assert arr.weights["1001"][0] == 0.50109E+02
    assert arr.weights["1002"][0] == 0.45746E+02
    assert arr.weights["1003"][0] == 0.52581E+02
    assert arr.weights["1004"][0] == 0.50109E+02
    assert arr.weights["1005"][0] == 0.45746E+02
    assert arr.weights["1006"][0] == 0.52581E+02
    assert arr.weights["1007"][0] == 0.50109E+02
    assert arr.weights["1008"][0] == 0.45746E+02
    assert arr.weights["1009"][0] == 0.52581E+02
```

Reply:
LGTM too. I will just add this explicit test of the values to make sure the weights are right.